graylog2-shared/src/main/java/org/graylog2/shared/plugins/PluginLoader.java

```diff
@@ -39,12 +48,17 @@ public PluginLoader(File pluginDir) {
     }
 
     public Set<Plugin> loadPlugins() {
-        final ImmutableSet.Builder<Plugin> plugins = ImmutableSet.builder();
+        final Set<Plugin> sorted = ImmutableSortedSet.orderedBy(new PluginComparator())
+                .addAll(loadClassPathPlugins())
+                .addAll(loadJarPlugins())
+                .build();
```

Comment:
Will this always add the classpath plugins before the jar plugins and always load the oldest plugin version?

I think it should be the other way around, allowing jar plugins to overlay classpath one's and later plugin versions to be preferred.